eclipse-theia/theia
# Validate Preferences Used in Editor

#### What it does

Fixes #10599 by adding a service to validate preference values used by Monaco.
Fixes #10678 by modifying the conditions for emitting events and the `affects` method of `PreferenceProxies`

It also improves the typing of JSON schemas and fixes some minor issues related to those typings.

#### How to test


1. start the application with theia as a workspace.
2. open the dev console to view logs.
3. open the settings.json file for the user or workspace scope.
4. add "editor.tabSize": 2 if it doesn't already exist - the application should work fine.
5. update the value to "editor.tabSize": "true".
6. Observe that application continues working normally as though the preference were set to its default value. There should also be a warning in the console describing the problem and explaining what value is being used.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
